src/uct/ib/base/ib_iface.c

```diff
@@ -850,19 +946,15 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
         goto err;
     }
 
-    status = uct_ib_device_select_gid_index(dev, self->config.port_num,
-                                            ib_md->config.gid_index,
-                                            &self->config.gid_index);
+    status = uct_ib_device_set_gid_info(dev, self->config.port_num,
+                                        ib_md->config.gid_index,
+                                        &self->gid_info);
     if (status != UCS_OK) {
         goto err;
     }
 
-    status = uct_ib_device_query_gid(dev, self->config.port_num,
-                                     self->config.gid_index, &self->gid,
-                                     &self->is_roce_v2);
-    if (status != UCS_OK) {
-        goto err;
-    }
+    self->is_roce_v2 = uct_ib_device_is_port_roce(dev, self->config.port_num) &&
```

Comment:
need to remove the field is_roce_v2 and use self->gid_info.roce_ver.ver directly when needed (or add helper function if used multiple times)